packages/forma-36-react-components/src/components/ControlledInput/ControlledInput.tsx

```diff
@@ -34,72 +34,74 @@ const defaultProps: Partial<ControlledInputPropTypes> = {
   willBlurOnEsc: true,
 };
 
-export class ControlledInput extends Component<ControlledInputPropTypes> {
-  static defaultProps = defaultProps;
+export const ControlledInput = (props: ControlledInputPropTypes) => {
+  const {
+    className,
+    id,
+    testId,
+    required,
+    disabled,
+    onFocus,
+    onBlur,
+    name,
+    onChange,
+    checked,
+    value,
+    type,
+    labelText,
+    willBlurOnEsc,
+    ...otherProps
+  } = props;
 
-  handleKeyDown = (e: KeyboardEvent<HTMLInputElement>) => {
-    const ESC = 27;
+  const classNames = cn(styles['ControlledInput'], className, {
+    [styles['ControlledInput--disabled']]: disabled,
+  });
 
-    if (e.keyCode === ESC && this.props.willBlurOnEsc) {
-      e.currentTarget.blur();
-    }
-  };
+  const handleKeyDown = useCallback(
+    (e: KeyboardEvent<HTMLInputElement>) => {
+      e.persist();
+      const ESC = '27';
 
-  render() {
-    const {
-      className,
-      id,
-      testId,
-      required,
-      disabled,
-      onFocus,
-      onBlur,
-      name,
-      onChange,
-      checked,
-      value,
-      type,
-      labelText,
-      willBlurOnEsc,
-      ...otherProps
-    } = this.props;
+      if (e.nativeEvent.code === ESC && willBlurOnEsc) {
```

Comment:
My bad, `code` works a bit differently than `keyCode`, as it checks against the key name instead of a number. See https://github.com/contentful/forma-36/blob/986eabd3d981818d8a3e35fb03b04132bcbd6033/packages/forma-36-react-components/src/components/Dropdown/Dropdown.tsx#L215

That's actually why I like it, since the condition is easier to read 🙂

Reply:
Ah, I should have tested it, cool though, it's clean. 👍